server/homebrew.api.js

```diff
@@ -209,6 +250,81 @@ const api = {
 
 		res.status(200).send(saved);
 	},
+	getThemeBundle : async(req, res)=>{
+		/*
+			getThemeBundle: Collects the theme and all parent themes
+				returns an object containing an array of css, in render order, and an array
+					of snippets ( currently empty )
+			Important parameter members:
+				req.params.id: This is the shareId ( User theme ) or name ( static theme )
+					loaded first.
+				req.params.renderer: This is the Markdown+ version for the static theme. If a
+					User theme the value will come from the User Theme metadata.
+		*/
+		req.params.renderer = _.upperFirst(req.params.renderer);
+		let currentTheme;
+		const completeStyles   = [];
+		const completeSnippets = [];
+
+		while (req.params.id) {
+			//=== User Themes ===//
+			if(!isStaticTheme(req.params.renderer, req.params.id)) {
+				await api.getBrew('share')(req, res, ()=>{});
+				currentTheme = req.brew;
+				splitTextStyleAndMetadata(currentTheme);
+
+				// If there is anything in the snippets or style members, append them to the appropriate array
+				if(currentTheme?.snippets) completeSnippets.push(JSON.parse(currentTheme.snippets));
+				if(currentTheme?.style) completeStyles.push(`/* From Brew: ${req.protocol}://${req.get('host')}/share/${req.params.id} */\n\n${currentTheme.style}`);
+
+				req.params.id       = currentTheme.theme;
+				req.params.renderer = currentTheme.renderer;
+			}
+			//=== Static Themes ===//
+			else {
+				const localSnippets = `${req.params.renderer}_${req.params.id}`; // Just log the name for loading on client
+				const localStyle    = `@import url(\"/themes/${req.params.renderer}/${req.params.id}/style.css\");`;
+				completeSnippets.push(localSnippets);
+				completeStyles.push(`/* From Theme ${req.params.id} */\n\n${localStyle}`);
+
+				req.params.id = Themes[req.params.renderer][req.params.id].baseTheme;
+			}
+		}
+
+		const returnObj = {
+			// Reverse the order of the arrays so they are listed oldest parent to youngest child.
+			styles   : completeStyles.reverse(),
+			snippets : completeSnippets.reverse()
+		};
+
+		res.setHeader('Content-Type', 'text/json');
+		return res.status(200).send(JSON.stringify(returnObj));
+	},
+	//Return CSS for a brew theme, with @include endpoint for its parent theme if any
+	getBrewThemeCSS : async (req, res)=>{
```

Comment:
So, with the rework to use `/theme`, is there a purpose for these separate `/css` endpoints anymore?

I don't think they are really needed anymore. *BUT*, now that `/theme` gets the whole chain of themes, maybe the niche for `/css` could be simplified to just retrieve the current brew styling and not the whole chain. Basically what https://github.com/naturalcrit/homebrewery/pull/3075 does (but support fetching styles from staticThemes too).

I'm curious what @ericscheid thinks, as he raised the original issue about a CSS endpoint https://github.com/naturalcrit/homebrewery/issues/1097

Reply:
Yes, #3075 is desired. It's a simple way to patch in one or more modules of css styling.

Reply:
And to be clear, you don't think that need is filled by this PR's `/theme` endpoint?